### ISSUE TITLE
Bug in MetricsTracker

### ISSUE THREAD

**Issue by coolmaksat** [keras-team/keras-tuner#86]
https://github.com/keras-team/keras-tuner/blob/0f785bce3d7c7a517c980efc6a8cfeb332f07581/kerastuner/engine/metrics_tracking.py#L67

Hi, I found a bug here in MetricsTracker's update function. The validation loss
of the best model returned by tuner.get_best_model with the best validation loss in
tuner.results_summary() was different. The reason is that the update function
returns true if the current value is <= than the max value in history. This should be
<= than the min value in history.
